dali/pipeline/operators/resize/resize_base.cc

```diff
@@ -104,13 +164,24 @@ void ResizeBase::Initialize(int num_threads) {
   resample_params_.resize(num_threads);
 }
 
-void ResizeBase::InitializeGPU() {
-  kernel_data_.resize(1);
-  auto &kdata = GetKernelData();
+void ResizeBase::InitializeGPU(int batch_size, int mini_batch_size) {
+  DALI_ENFORCE(batch_size > 0, "Batch size must be positive");
+  DALI_ENFORCE(mini_batch_size > 0, "Mini-batch size must be positive");
```

Comment:
shouldn't you enforce that `mini_batch_size` is actually smaller than `batch_size`?

Reply:
This is a bit of a guideline. E.g. if you specify minibatch of 32 and batch of 208, then you get some minibatches of 29, some of 30 (and none 32). If batch_size < minbatch_size, then you get just 1 minibatch.